packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -214,37 +296,73 @@ export class CodeSnippetWidget extends ReactWidget {
   codeSnippetManager: CodeSnippetService;
   renderCodeSnippetsSignal: Signal<this, ICodeSnippet[]>;
   getCurrentWidget: () => Widget;
+  app: JupyterFrontEnd;
+  editorServices: IEditorServices;
 
-  constructor(getCurrentWidget: () => Widget) {
+  constructor(
+    getCurrentWidget: () => Widget,
+    app: JupyterFrontEnd,
+    editorServices: IEditorServices
+  ) {
     super();
     this.getCurrentWidget = getCurrentWidget;
     this.codeSnippetManager = new CodeSnippetService();
     this.renderCodeSnippetsSignal = new Signal<this, ICodeSnippet[]>(this);
+    this.openCodeSnippetEditor = this.openCodeSnippetEditor.bind(this);
+    this.app = app;
+    this.editorServices = editorServices;
+
+    this.fetchData = this.fetchData.bind(this);
+    this.updateSnippets = this.updateSnippets.bind(this);
   }
 
   // Request code snippets from server
   async fetchData(): Promise<ICodeSnippet[]> {
     return await this.codeSnippetManager.findAll();
   }
 
-  // Triggered when the widget button on side palette is clicked
-  onAfterShow(msg: Message): void {
+  updateSnippets(): void {
     this.fetchData().then((codeSnippets: ICodeSnippet[]) => {
       this.renderCodeSnippetsSignal.emit(codeSnippets);
     });
   }
 
+  // Triggered when the widget button on side panel is clicked
+  onAfterShow(msg: Message): void {
+    this.updateSnippets();
+  }
+
+  addCodeSnippet(): void {
+    const newSnippetArgs = JSON.parse(JSON.stringify(properties.base_args));
+    newSnippetArgs.updateSignal = this.updateSnippets;
+    this.openCodeSnippetEditor(newSnippetArgs);
+  }
+
+  openCodeSnippetEditor(args: any): void {
+    this.app.commands.execute(`${METADATA_EDITOR_ID}:open`, args);
+  }
+
   render(): React.ReactElement {
     return (
       <div className={CODE_SNIPPETS_CLASS}>
         <header className={CODE_SNIPPETS_HEADER_CLASS}>
-          {'</> Code Snippets'}
+          <p> {'</> Code Snippets'} </p>
```

Comment:
I know this has been like this for a while, maybe we should use code snippet icon instead of `</>`?

Reply:
Just added issue #683 to do this (but I don't want to hold up this PR by trying to figure that out)